hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/utils/SparkMetadataWriterUtils.java

```diff
@@ -194,66 +207,21 @@ private static List<HoodieRecord> getBaseFileRecords(HoodieBaseFile baseFile, Ho
     }
   }
 
-  private static Dataset<Row> toDataset(List<HoodieRecord> records, Schema schema, SQLContext sqlContext, boolean isBaseFile) {
-    List<GenericRecord> avroRecords = records.stream()
+  private static List<Row> toRows(List<HoodieRecord> records, Schema schema, SQLContext sqlContext, String path) {
+    StructType structType = AvroConversionUtils.convertAvroSchemaToStructType(schema);
+    Function1<GenericRecord, Row> converterToRow = AvroConversionUtils.createConverterToRow(schema, structType);
+    List<Row> avroRecords = records.stream()
         .map(r -> {
-          if (isBaseFile) {
-            return (GenericRecord) r.getData();
-          }
-          HoodieRecordPayload payload = (HoodieRecordPayload) r.getData();
           try {
-            return (GenericRecord) payload.getInsertValue(schema).get();
+            return (GenericRecord) (r.getData() instanceof GenericRecord ? r.getData()
+                : ((HoodieRecordPayload) r.getData()).getInsertValue(schema, new Properties()).get());
```

Comment:
Passed empty properties here. How do we usually pass properties for this function?

Reply:
not sure I get your question. We pass it in from driver only (hoodieWriteConfig.getProps())

Reply:
Fixed it